### PR TITLE
add telemetry cm

### PR DESCRIPTION
**Pull request by hoegaarden**
/ref: TAPPC-126

### CODE REVIEW

File: tap-setup-scripts/src/resources/package_overlay_secrets.yml
Comment (satya-dillikar) [inline tap-setup-scripts/src/resources/package_overlay_secrets.yml]: cosmetic: rename: aws-quickstart ; as we will have azure-quickstart as well.

File: tap-setup-scripts/src/resources/package_overlay_secrets.yml
Comment (satya-dillikar) [inline tap-setup-scripts/src/resources/package_overlay_secrets.yml]: rename this file extension as ".yaml" not "yml" to be consistent with other files.

File: tap-setup-scripts/src/functions.sh
Comment (satya-dillikar) [inline tap-setup-scripts/src/functions.sh:410]: you might wish to use "requireValue" function for mandatory params

File: tap-setup-scripts/src/resources/package_overlay_secrets.yml
Comment (satya-dillikar) [inline tap-setup-scripts/src/resources/package_overlay_secrets.yml]: I guess the quickstartBuild value should be unique to the TAP-on-AWS-QS release. we should generate that UUID and monitor that in telemetry. 

I understand we don't have a fool-proof mechanisms to track QS customers in telemetry but this should be the starting point.

**Comment by hoegaarden** [comment]
rebased on https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/pull/13

File: tap-setup-scripts/src/resources/package_overlay_secrets.yml
Comment (hoegaarden) [reply tap-setup-scripts/src/resources/package_overlay_secrets.yml]: According to the story and the slack convo that @aaronmeza had, the name should be `quickstart`. As this is a feature that has to be implemented in the telemetry controller yet, I am not sure which CMs / which names they'd consider.

@aaronmeza ; do you have any input on that? If not, I'll just add to the slack thread you had started a while ago and try to figure that out with the telemetry folks.

File: tap-setup-scripts/src/resources/package_overlay_secrets.yml
Comment (hoegaarden) [reply tap-setup-scripts/src/resources/package_overlay_secrets.yml]: I don't think that should be a UUID, I think that should be a release version of the QS.
I imagine we'll have the QS eventually versioned and this would be that.

Quoting the story:
```
The key-value pair will be version={TAP_VERSION}+{platform}.{buildID}
Where the 3-character CSP platform abbreviation is used in build metadata.
For instance, for the AWS quickstart running TAP 1.4.0, the string would look like:
version=1.4.0+aws.5
```

For now, I set it to `0` because we essentially don't have proper versioning going yet.

File: tap-setup-scripts/src/functions.sh
Comment (hoegaarden) [reply tap-setup-scripts/src/functions.sh:410]: `requireValue`
- explicitly `exit`s (also on interactive sessions) and thus makes it somewhat weird to test / run in isolation
- does not seem to work with positional params

So I'd rather stick with the shell builtin, except there is some benefit of the function I am not aware of.

**Comment by hoegaarden** [comment]
Things to check / potentially for acceptance:
- in the bastion hosts `/var/log/cloud-init-output.log` there should be lines like:
  ```
     [...]
  ### Installing TAP values from /home/ubuntu/tap-setup-scripts/generated/tap-values-single.yaml for TAP_VERSION 1.3.0 ...

  ### Adding package overlays

  secret/tap-telemetry-overlays created

   Added installed package 'tap'
  ### Checking state of all packages
     [...]
  ```
- in the `tap-install` namespace
  - there should be a secret `tap-telemetry-overlays`
  - the `pkgi` `tap-telemetry` should have the overlay annotation using that secret
- in the `tap-telemetry` namespace
  - there should be a `configamp` `quickstart` (name might change according to the outcome of the [thread above](https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/pull/11#discussion_r1051048359)) with `data`:
    ```
    version: 1.3.0+aws.0
    ```

**Comment by satya-dillikar** [comment]
I have resolved all the comments. I approve the changes. let us wait for 2nd approver before the merge.

**Comment by hoegaarden** [comment]
Because it came up in the standup, calling this out explicitly:
- The build number is, for now, hardcoded to `0`.
- Once we have CI going, we can use get the build number (the git sha, or a github release or something) from there and pass that through

**Comment by hoegaarden** [comment]
closes: #13
this PR was initially based on #13 and superseds it.

**Review by aaronmeza — APPROVED** [review]

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [inline templates/aws-tap-entrypoint-existing-vpc.template.yaml:1461]: all these s3 cp are required. Is this an oversight?

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [inline templates/aws-tap-entrypoint-existing-vpc.template.yaml:805]: if this is already taken care of in the upstream repo then we can get it by the next merge/pull from upstream.

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:1461]: The individual copies have been replaced with a "wildcard copy", see: https://github.com/vmware-tap-on-public-cloud/quickstart-vmware-tanzu-application-platform/blob/410b83c573a54068d78fda0530f47418a1062164/templates/aws-tap-entrypoint-existing-vpc.template.yaml#L1462-L1465

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (hoegaarden) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:805]: This came in, as a cherry-pick, from the upstream repo / commit.
I don't see a problem if bring it in, and merge it; makes the diff between our `main` and the upstream a little smaller.
Am I missing what you are concerned about?

File: templates/aws-tap-entrypoint-existing-vpc.template.yaml
Comment (satya-dillikar) [reply templates/aws-tap-entrypoint-existing-vpc.template.yaml:1461]: I get it.  This is fine for now.
FYI, we have some files like git-ssh-basic-auth.yaml & workload-aws.example.yaml not used in the code.

**Review by satya-dillikar — APPROVED** [review]

**Pull request merged**
